### ISSUE TITLE
`/usr/lib/qubes/cleanup-dispvms` hangs

### ISSUE THREAD

**Issue by DemiMarie** [QubesOS/qubes-issues#8619]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

R4.2

### Brief summary

`/usr/lib/qubes/cleanup-dispvms` hangs

### Steps to reproduce

Not sure.  This happened once for me.

### Expected behavior

It doesn’t hang, or at least does not hang the boot process.

### Actual behavior

Freezes and hangs the boot process.

**Comment by marmarek** [comment]
Which dom0 kernel? If 6.5, it might be #8575

**Comment by DemiMarie** [comment]
> Which dom0 kernel? If 6.5, it might be #8575

`uname -r` gives `6.1.57-1.qubes.fc37.x86_64`.

**Comment by rustybird** [comment]
Did you by any chance boot with `preempt=full` when it happened?

**Comment by DemiMarie** [comment]
> Did you by any chance boot with `preempt=full` when it happened?

Yes, why do you ask?

**Comment by rustybird** [comment]
Just a hunch. No smoking gun yet :)

I've had what might have been a similar kind of hang like @marmarek's in #8575 a week or two ago, on shutting down a VM while lots of I/O was going on elsewhere. Unfortunately I didn't trigger a sysrq "Show Blocked State" back then to compare. His does involve dm-crypt. My system (which is on kernel-latest 6.5) uses Btrfs not LVM; maybe LVM is just a bit more likely, if even that, to trigger some bug in the device-mapper or a lower layer of the storage stack?

And so I've been wondering if the scheduler change could make this hypothetical bug more likely to cause an actual hang, although maybe that's because the recent LWN article series on reworking the kernel's preemption code ([part 1](https://lwn.net/Articles/944686/), [part 2](https://lwn.net/Articles/945422/)) has primed me to see it as a potential source of trouble.

**Comment by DemiMarie** [comment]
@rustybird I don’t think this was a kernel deadlock.  `systemctl restart qubesd.service` solved the problem.

**Comment by rustybird** [comment]
So `qubes-core.service` was hanging during boot before login, and running `systemctl restart qubesd.service` (how?) unwedged it?

**Comment by DemiMarie** [comment]
Duplicate of #8754 which has more information as to the actual cause.
